extensions/netty/runtime/src/main/java/io/quarkus/netty/runtime/graal/NettySubstitutions.java

```diff
@@ -561,6 +565,33 @@ protected EmbeddedChannel newContentDecompressor(ChannelHandlerContext ctx, Char
     }
 }
 
+@TargetClass(className = "io.netty.handler.codec.http2.CompressorHttp2ConnectionEncoder")
+final class Target_CompressorHttp2ConnectionEncoder {
+    @Delete
+    private BrotliOptions brotliOptions;
+    @Delete
+    private ZstdOptions zstdOptions;
+
+    @Alias
+    @KeepOriginal
+    private EmbeddedChannel newCompressionChannel(final ChannelHandlerContext ctx, ZlibWrapper wrapper) {
+        return null;
+    }
+
+    @Substitute
+    protected EmbeddedChannel newContentCompressor(ChannelHandlerContext ctx, CharSequence contentEncoding)
+            throws Http2Exception {
+        if (GZIP.contentEqualsIgnoreCase(contentEncoding) || X_GZIP.contentEqualsIgnoreCase(contentEncoding)) {
+            return newCompressionChannel(ctx, ZlibWrapper.GZIP);
+        }
+        if (DEFLATE.contentEqualsIgnoreCase(contentEncoding) || X_DEFLATE.contentEqualsIgnoreCase(contentEncoding)) {
+            return newCompressionChannel(ctx, ZlibWrapper.ZLIB);
+        }
```

Comment:
I would throw UnsupportedOperationException if brotli or the other one are used. Otherwise, it will run as identity (which would be wrong) and the user won't have any clue.